internal/datanode/importv2/memory_allocator.go

```diff
@@ -96,3 +111,23 @@ func (ma *memoryAllocator) Release(taskID int64, size int64) {
 		zap.Int64("releasedSize", size),
 		zap.Int64("usedMemory", ma.usedMemory))
 }
+
+// BlockingAllocate blocks until memory is available and then allocates
+func (ma *memoryAllocator) BlockingAllocate(taskID int64, size int64) {
+	// First try to allocate immediately
+	if ma.tryAllocate(taskID, size) {
+		return
+	}
+
+	log.Info("task waiting for memory allocation",
+		zap.Int64("taskID", taskID),
+		zap.Int64("requestedSize", size))
+
+	// Keep trying until allocation succeeds
+	for {
+		time.Sleep(5 * time.Second)
```

Comment:
do  not use sleep, use conditionvariable instead.

Reply:
done